### ISSUE TITLE
Bad output of airmon-ng

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#17]
**Reported by misterx on 16 Dec 2006 22:26 UTC**

This happens on ubuntu (and maybe other distro) with [airmon-ng <start|stop> <interface> [channel](18]:

```
usage:)
 
-e Interface    Chipset         Driver
 
-e -n wifi0             Atheros         madwifi-ng
 
[: 377: ==: unexpected operator
-e -n ath0              Atheros         madwifi-ng VAP (parent: wifi0)
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Dec 2006 22:26 UTC**

This could be fixed by using /bin/bash instead of /bin/sh like it was done in a previous revision but it breaks ash compatibility.

By default, we will use /bin/bash but if it doesn't exist, we will search for /bin/sh and if this one doesn't exist, we will use the current shell that can be found with $SHELL (at least on bash).

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Dec 2006 22:26 UTC**

That's temporarly fixed for most distro in #25

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Dec 2006 22:26 UTC**

(In #26) Some improvment to makefiles and under linux, it should detect the shell used (Closes: #17)
